### PR TITLE
Travis CI: Add parallel test runs on macOS and Windows

### PR DESCRIPTION
**Pull request by cclauss**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

Travis CI: Add parallel test runs on macOS and Windows

Windows and Python 3 run in [__allow_failures__ mode](https://docs.travis-ci.com/user/build-matrix/#rows-that-are-allowed-to-fail) until we are ready to support them.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
+      before_install: HOMEBREW_NO_AUTO_UPDATE=1 brew install npm
+    - name: "Python 2.7 on Windows"
+      os: windows
+      language: node_js
+      node_js: 12  # node
+      env: PATH=/c/Python27:/c/Python27/Scripts:$PATH
+      before_install: choco install python2
+  allow_failures:
+    - os: windows
+    - python: 3.7
```
Comment (rvagg) [inline .travis.yml]: are we expecting failures on windows now that #1793 is landed?

**Comment by rvagg** [comment]
both of these commits should probably be prefixed with `test:`

File: .travis.yml (hunk 1 above)
Comment (cclauss) [reply .travis.yml]: Python 3 yes but Windows no.  See comments below.

**Comment by rvagg** [comment]
Oh, and here's another thing, **avoid merge commits**, they mess up the sequential flow of commits on a branch and make the history hard to traverse. Only use `git merge` when you **know** that you're merging a branch that you know is up to date. If you end up being asked to write a commit message for `git merge` then you've done it wrong.

So what I do in cases like this goes something like this:

```
git checkout master
git pull origin master
git checkout my-local-pull-request-branch
git rebase master
git push origin my-local-pull-request-branch --force
```

and you can shorten that a bit and avoid the master checkout while staying on your local pull request's branch:

```
git fetch origin
git rebase origin/master
git push origin my-local-pull-request-branch --force
```

Fixing this branch might be a bit awkward now. What I'd probably do to land it is `cherry-pick` the first two pre-merge commits onto a fresh branch, then cherry-pick the post-merge one, fixing any conflicts manually.

If you're happy with this I could probably do all that and land it if you'd like.
There's failures on Windows but I don't suppose it's the fault of this PR: https://travis-ci.com/nodejs/node-gyp/jobs/211035846

**Comment by cclauss** [comment]
Understood.  My git skills are not all that they should be so I often rely on GitHub web UI which I will stop doing on Nodejs projects.  I use the cheatsheet for CPython that has a similar approach to yours: https://devguide.python.org/gitbootcamp/?highlight=upstream#syncing-with-upstream

Please go ahead and do the manipulations to land this one when you are OK with it.  The failure on Windows can be addressed in future PRs.

**Comment by rvagg** [comment]
squashed into one commit and landed into b93bac9

**Comment by rvagg** [comment]
sorry, 7a9a038

**Pull request closed without merging**